builtin/providers/aws/resource_aws_opsworks_user_profile.go

```diff
@@ -0,0 +1,135 @@
+package aws
+
+import (
+	"log"
+
+	"github.com/hashicorp/terraform/helper/schema"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/opsworks"
+)
+
+func resourceAwsOpsworksUserProfile() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsOpsworksUserProfileCreate,
+		Read:   resourceAwsOpsworksUserProfileRead,
+		Update: resourceAwsOpsworksUserProfileUpdate,
+		Delete: resourceAwsOpsworksUserProfileDelete,
+
+		Schema: map[string]*schema.Schema{
+			"id": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"user_arn": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+
+			"allow_self_management": &schema.Schema{
+				Type:     schema.TypeBool,
+				Optional: true,
+			},
+
+			"ssh_username": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+
+			"ssh_public_key": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+		},
+	}
+}
+
+func resourceAwsOpsworksUserProfileRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*AWSClient).opsworksconn
+
+	req := &opsworks.DescribeUserProfilesInput{
+		IamUserArns: []*string{
+			aws.String(d.Id()),
+		},
+	}
+
+	log.Printf("[DEBUG] Reading OpsWorks user profile: %s", d.Id())
+
+	resp, err := client.DescribeUserProfiles(req)
+	if err != nil {
+		if awserr, ok := err.(awserr.Error); ok {
+			if awserr.Code() == "ResourceNotFoundException" {
+				log.Printf("[DEBUG] OpsWorks user profile (%s) not found", d.Id())
+				d.SetId("")
+				return nil
+			}
+		}
+		return err
+	}
+
+	for _, profile := range resp.UserProfiles {
+		d.Set("allow_self_management", profile.AllowSelfManagement)
+		d.Set("user_arn", profile.IamUserArn)
+		d.Set("ssh_public_key", profile.SshPublicKey)
+		d.Set("ssh_username", profile.SshUsername)
+		break
+	}
+
+	return nil
+}
+
+func resourceAwsOpsworksUserProfileCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*AWSClient).opsworksconn
+
+	req := &opsworks.CreateUserProfileInput{
+		AllowSelfManagement: aws.Bool(d.Get("allow_self_management").(bool)),
```

Comment:
As this is an optional param, with no default, what will happen here if there is no value set?